catroid/src/main/java/org/catrobat/catroid/formulaeditor/InternToExternGenerator.java

```diff
@@ -319,6 +313,41 @@ private String generateExternStringFromToken(InternToken internToken, boolean tr
 		}
 	}
 
+	private String generateExternNumberStringForToken(String number, boolean trimNumbers) {
```

Comment:
![dkur9tk](https://user-images.githubusercontent.com/15098185/35867929-01ca6748-0b53-11e8-9149-916f6cfb5f9a.png)

maybe if i sum this up again, youll see why:
- you get a string
- only if the number is to be trimmed, you create an anonymous instance of an Formula Brick
- for the sole purpose of parsing the string to a float
- to then call a public method of formulaBrick
- thats just there to give you a hadle for a protected method
- to then check what the method actually did with it by checking if theres a "." in it.

and all this to just format a number string.
please check on what is the expected behaviour of this in the first place and then write a method here. it seems like you write a whole lot of code, just for the sake of beeing able to use a method from FormulaBrick while still spending a lot of effort to make that return value fit your needs. and all that while breaking the idea of access modifiers..